.azure-pipelines/ultimate-pipeline.yml

```
@@ -3987,8 +3987,13 @@ stages:
       continueOnError: true
 
 - stage: throughput_appsec
-  condition: and(succeeded(), eq(variables.isMainRepository, true))
-  dependsOn: [package_linux, merge_commit_id]
+  condition: >
+    and(succeeded(), 
+        eq(variables.isMainRepository, true),
+        or(
+          eq(variables.isMainBranch, true),
```

You may want to have the ability to force an appsec run by setting a variable. We have this for other stages. e.g. 
```suggestion
          eq(variables.isMainBranch, true),
          eq(variables.force_appsec_throughput_run, 'true'),
```